tests/api/test_incidents.py

```diff
@@ -0,0 +1,138 @@
+from django.urls import reverse
+from faker import Faker
+from rest_framework.test import force_authenticate
+import json
+import pytest
+import random
+
+from response.models import Incident, ExternalUser
+from response import serializers
+from response.core.views import IncidentViewSet
+from tests.factories import IncidentFactory
+
+faker = Faker()
+
+
+def test_list_incidents(arf, api_user):
+    persisted_incidents = IncidentFactory.create_batch(5)
+
+    req = arf.get(reverse("incident-list"))
+    force_authenticate(req, user=api_user)
+    response = IncidentViewSet.as_view({"get": "list"})(req)
+
+    assert response.status_code == 200, "Got non-200 response from API"
+    content = json.loads(response.rendered_content)
+
+    assert "results" in content, "Response didn't have results key"
+    incidents = content["results"]
+    assert len(incidents) == len(
+        persisted_incidents
+    ), "Didn't get expected number of incidents back"
+
+    for idx, incident in enumerate(incidents):
+        assert incident["report_time"]
+
+        # incidents should be in order of newest to oldest
+        if idx != len(incidents) - 1:
+            assert (
+                incident["report_time"] >= incidents[idx + 1]["report_time"]
+            ), "Incidents are not in order of newest to oldest by report time"
+
+        assert "end_time" in incident  # end_time can be null for open incidents
+        assert incident["impact"]
+        assert incident["report"]
+        assert incident["start_time"]
+        assert incident["summary"]
+        assert incident["severity"]
+
+        reporter = incident["reporter"]
+        assert reporter["display_name"]
+        assert reporter["external_id"]
+
+        lead = incident["lead"]
+        assert lead["display_name"]
+        assert lead["external_id"]
+
+        # TODO: verify actions are serialised inline
+
+
+@pytest.mark.parametrize(
+    "update_key,update_value",
+    (
+        ("", ""),  # no update
+        ("impact", faker.paragraph(nb_sentences=2, variable_nb_sentences=True)),
```

Comment:
Use [faker](https://faker.readthedocs.io/en/stable/) to generate random text for use here (see comment on factories below)